src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -19,9 +19,12 @@ def index
   end
 
   def update
+    # If we filter by "read", we want to "unread" the notifications, ie. undeliver them.
+    # In any other case, we display `Mark as "read"`, indicating we want to make all
+    # the selected notifications be delivered.
+    delivered = params[:state] == 'read'
     # rubocop:disable Rails/SkipsModelValidations
-    @read_count = Notification.where(id: @undelivered_notification_ids).update_all('delivered = !delivered')
-    @unread_count = Notification.where(id: @delivered_notification_ids).update_all('delivered = !delivered')
+    @count = @notifications.where(id: @notification_ids).where(delivered: delivered).update_all(delivered: !delivered)
```

Comment:
```suggestion
    @count = @notifications.where(id: @notification_ids, delivered: delivered).update_all(delivered: !delivered)
```
Do we need to chained `.where` conditions?

Reply:
```
[1] pry(main)> Notification.where(id: 1).where(delivered: false).explain
  Notification Load (0.4ms)  SELECT `notifications`.* FROM `notifications` WHERE `notifications`.`id` = 1 AND `notifications`.`delivered` = FALSE
=> EXPLAIN for: SELECT `notifications`.* FROM `notifications` WHERE `notifications`.`id` = 1 AND `notifications`.`delivered` = FALSE
+----+-------------+---------------+-------+------------------------------------------+---------+---------+-------+------+-------+
| id | select_type | table         | type  | possible_keys                            | key     | key_len | ref   | rows | Extra |
+----+-------------+---------------+-------+------------------------------------------+---------+---------+-------+------+-------+
|  1 | SIMPLE      | notifications | const | PRIMARY,index_notifications_on_delivered | PRIMARY | 8       | const | 1    |       |
+----+-------------+---------------+-------+------------------------------------------+---------+---------+-------+------+-------+
1 row in set (0.00 sec)

[2] pry(main)> Notification.where(id: 1, delivered: false).explain
  Notification Load (0.7ms)  SELECT `notifications`.* FROM `notifications` WHERE `notifications`.`id` = 1 AND `notifications`.`delivered` = FALSE
=> EXPLAIN for: SELECT `notifications`.* FROM `notifications` WHERE `notifications`.`id` = 1 AND `notifications`.`delivered` = FALSE
+----+-------------+---------------+-------+------------------------------------------+---------+---------+-------+------+-------+
| id | select_type | table         | type  | possible_keys                            | key     | key_len | ref   | rows | Extra |
+----+-------------+---------------+-------+------------------------------------------+---------+---------+-------+------+-------+
|  1 | SIMPLE      | notifications | const | PRIMARY,index_notifications_on_delivered | PRIMARY | 8       | const | 1    |       |
+----+-------------+---------------+-------+------------------------------------------+---------+---------+-------+------+-------+
1 row in set (0.00 sec)
```
Regarding performance, the query it does its the same, so that's maybe a matter of style...